lib/openapi.js

```diff
@@ -0,0 +1,262 @@
+          swaggerMethod.requestBody = {
+            content: {}
+          }
+          getBodyParams(swaggerMethod.requestBody.content, schema.body, schema.consumes)
+        }
+
+        if (schema.params) {
+          getPathParams(parameters, schema.params)
+        }
+
+        if (schema.headers) {
+          getHeaderParams(parameters, schema.headers)
+        }
+
+        if (parameters.length) {
+          swaggerMethod.parameters = parameters
+        }
+
+        if (schema.deprecated) {
+          swaggerMethod.deprecated = schema.deprecated
+        }
+
+        if (schema.security) {
+          swaggerMethod.security = schema.security
+        }
+
+        if (schema.servers) {
+          swaggerMethod.servers = schema.servers
+        }
+
+        for (const key of Object.keys(schema)) {
+          if (key.startsWith('x-')) {
+            swaggerMethod[key] = schema[key]
+          }
+        }
+      }
+
+      swaggerMethod.responses = genResponse(schema ? schema.response : null)
+
+      swaggerObject.paths[url] = swaggerRoute
+    }
+
+    if (opts && opts.yaml) {
+      const swaggerString = yaml.safeDump(swaggerObject, { skipInvalid: true })
+      cache.swaggerString = swaggerString
+      return swaggerString
+    }
+
+    cache.swaggerObject = swaggerObject
+    return swaggerObject
+
+    function getBodyParams (parameters, body, consumes) {
+      const bodyResolved = ref.resolve(body)
+
+      if ((Array.isArray(consumes) && consumes.length === 0) || typeof consumes === 'undefined') {
+        consumes = ['application/json']
+      }
+
+      consumes.forEach((consume) => {
+        parameters[consume] = {
+          schema: bodyResolved
+        }
+      })
+    }
+
+    function getQueryParams (parameters, query) {
+      const resolved = ref.resolve(query)
+      const add = plainJsonObjectToSwagger2('query', resolved, swaggerObject.components.schemas)
+      add.forEach(_ => parameters.push(swagger2ParametersToOpenapi3(_)))
+    }
+
+    function getPathParams (parameters, path) {
+      const resolved = ref.resolve(path)
+      const add = plainJsonObjectToSwagger2('path', resolved, swaggerObject.components.schemas)
+      add.forEach(_ => parameters.push(swagger2ParametersToOpenapi3(_)))
+    }
+
+    function getHeaderParams (parameters, headers) {
+      const resolved = ref.resolve(headers)
+      const add = plainJsonObjectToSwagger2('header', resolved, swaggerObject.components.schemas)
+      add.forEach(_ => parameters.push(swagger2ParametersToOpenapi3(_)))
+    }
+
+    // https://swagger.io/docs/specification/2-0/describing-responses/
+    function genResponse (fastifyResponseJson) {
+      // if the user does not provided an out schema
+      if (!fastifyResponseJson) {
+        return { 200: { description: 'Default Response' } }
+      }
+
+      const responsesContainer = {}
+
+      Object.keys(fastifyResponseJson).forEach(key => {
+        // 2xx is not supported by swagger
```

Comment:
Can you elaborate?

Reply:
I think this one is swagger's legacy. Openapi does support `2XX`.

Reply:
Forget to remove the invalid comment.